src/ir/auth_logic/lowering_ast_datalog.cc

```diff
@@ -195,7 +203,9 @@ LoweringToDatalogPass::FactToDLIR(const Principal& speaker, const Fact& fact) {
     datalog::Predicate prin_cansay_pred =
         PushPrincipal("canSay_", delegatees, inner_fact_dlir);
     inner_fact_dlir = prin_cansay_pred;
+    can_say_depth_inner += 1;
```

Comment:
Doesn't this logic just make `can_say_depth_inner` have the same value as `fact.delecation_chain().size()`? Would it be more concise to just pass that into `UpdateCanSayDepth`?

Reply:
You are correct. That would eliminate introducing a new variable. I will add some more tests around can_say_depth just to be safe here.